crypto/o_str.c

```diff
@@ -90,6 +90,58 @@ size_t OPENSSL_strlcat(char *dst, const char *src, size_t size)
     return l + OPENSSL_strlcpy(dst, src, size);
 }
 
+/**
+ * @brief Converts a string to an unsigned long integer.
```

Comment:
Should probably document that `errno` is reset to 0 in the case of no error, or set if there is an error.

Or could say that this is a wrapper around the system `strtoul`, which would then imply this (but nothing in the doc says this function is implemented that way, and the doc should say if `errno` is touched)